### PR TITLE
Fix DI.Helpers.sendLog/sendClyde

### PR DESCRIPTION
**Pull request by TTtie**
- `DI.Helpers.sendClyde` uses real Discord Clyde instead of a mock through `DI.Helpers.sendLog`
- All of the mentioned methods use Discord's internal methods to send the message; for custom avatars the style is still edited. Also that means, the methods don't depend on `DI.ws` now!

### CODE REVIEW

File: DomReady/Helpers.js [hunk 1]
```diff
@@ -68,26 +42,22 @@ class Helpers {
...
-
-        output.data = JSON.stringify(output.data);
+                }
         return output;
     }
 
     sendClyde(message) {
-        return this.sendLog('Clyde', message, undefined);
+        let a = new Date().getTime()
+        webpackJsonp([],{[a]:(a,b,d)=>{d(56).sendBotMessage(window.DI.client.selectedChannel.id, message)}},[a])
```
Comment (cking) [inline DomReady/Helpers.js]: Get a discord update, ID breaks. In the constructor, search for the ID dynamically pls.

File: DomReady/Helpers.js [hunk 2]
```diff
@@ -108,23 +79,20 @@ class Helpers {
...
-        }
-        document.querySelector('.messages').scrollTop = elem.offsetTop;
+        webpackJsonp([],{[a]:(a,b,d)=>{d(56).receiveMessage(window.DI.client.selectedChannel.id, this.constructMessage(base))}},[a])
+        const className="is-local-bot-message"
+        let elem = document.querySelector(`.${className}:last-child .avatar-large`);
+        elem.setAttribute('style', `background-image: url('${avatarURL}');`);
+        const destroyMessage = () => {
+            try {
+                document.querySelector(`.${className}:last-child .comment .local-bot-message a`).click()
+                DI.StateWatcher.removeListener("channelChanged", destroyMessage) // fuck eventemitter3
```
Comment (cking) [inline DomReady/Helpers.js]: A) any alternative?
B) no swear words pls.

File: DomReady/Helpers.js [hunk 3]
```diff
@@ -108,23 +79,20 @@ class Helpers {
...
+            } catch(e) {
+                if (e.message === "Cannot read property 'click' of null") return // js is cryïng
```
Comment (cking) [inline DomReady/Helpers.js]: No special chars (your i has 2 dots)

File: DomReady/Helpers.js [hunk 4]
```diff
@@ -108,23 +79,20 @@ class Helpers {
...
+                throw e;
+            }
+        } 
+        DI.StateWatcher.on("channelChanged", destroyMessage) // Destroy the message because I'm lazy to fix one bug
```
Comment (cking) [inline DomReady/Helpers.js]: If there is a known bug, document it!!!

**Comment by Ratismal** [comment]
testing in #68

**Comment by Ratismal** [comment]
looks good imo, the only issue i encountered is that if two different fake messages are sent consecutively (not real ones), all of them in that group get conformed to the last message sent. idk if i really care enough to go back to the awful system i had before though.

@cking any last updates to your review before I merge?

**Comment by cking** [comment]
lgtm

**Comment by TTtie** [comment]
only a small **TODO**: *document all mentioned issues in this PR somewhere, ie. `KNOWN_ISSUES.md`*

**Comment by cking** [comment]
Do it ;)

**Pull request merged**